### PR TITLE
Fixed issue search, it now works

### PR DESCRIPTION
**Pull request by KrauseFx**
- Fixed not actually filtering the repo
- Use GitHub magic order of results
- Fixed URL escaping
- Prepare to make sort_by and order configurable in the future
- Logging of search URL in verbose mode

### CODE REVIEW

File: lib/sidekick.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def search(query, delegate)
       url = url_for_request query
 
       begin
-        results = get_api_results url
+        results = get_api_results(url)
```
Comment (orta) [inline lib/sidekick.rb]: heh

File: lib/sidekick.rb [hunk 2]
```diff
@@ -51,20 +51,27 @@ def verbose
     require 'json'
 
     # Generates a URL for the request
-    def url_for_request(query)
-      "https://api.github.com/search/issues?q=#{URI.escape query}&repo=#{repo_owner}/#{repo_name}&sort=created&order=asc"
+    def url_for_request(query, sort_by: nil, order: nil)
```
Comment (orta) [inline lib/sidekick.rb]: When I compared crashes to search results, the sort_by and order were pretty important in deciding, especially when there was a few similar results

Comment (orta) [inline lib/sidekick.rb]: But after checking the fastlane PR, looks like things worked well there

**Comment by KrauseFx** [comment]
🚀

**Pull request merged**